### DISCUSSION TITLE
Get  PointerPosition

### DISCUSSION THREAD

**Discussion by steam3d** [microsoft/WindowsAppSDK#3877] [Q&A] [answered]
How to get  PointerPosition when app starting?  PointerMoved will not fire while user does not move a pointer at least on 1px, other using CoreWindow.PointerPosition also does not work for Winui3

**Comment by DarranRowe** [comment]
[GetCursorPos](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-getcursorpos) or [Cursor.Position](https://learn.microsoft.com/en-us/dotnet/api/system.windows.forms.cursor.position?view=windowsdesktop-7.0&redirectedfrom=MSDN#System_Windows_Forms_Cursor_Position) should get you the current mouse cursor position or potentially the pen position if it is hovering and in range. I say potentially, because I don't know for sure how pen input interacts. But it looks like it takes over the mouse cursor.
Because of how touch works, you can never get any current position until it makes contact, and then you are guaranteed to get the PointerEntered event which gives the first position for that pointer.

What I gave are Windows desktop functions. That is because everything in the Windows App SDK is a library that runs in a desktop process.

**Reply by steam3d** [reply]
I think about GetCursorPos but it give the global point. I have no idea how to convert it to local point, because I want to bind image to cursor.

**Reply by steam3d** [reply]
I also think that I can move mouse to 1px to get PointerMoved event

**Reply by DarranRowe** [reply, marked as answer]
Yes, the GetCursorPos documentation states:
"Retrieves the position of the cursor, **in screen coordinates**."
This is why [ScreenToClient](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-screentoclient) exists. Remember, because the mouse cursor is a system service, you have to be prepared for the situation where the cursor is outside of your window.
